modules/apps/portal/portal-messaging/src/main/java/com/liferay/portal/messaging/internal/DefaultMessageBus.java

```diff
@@ -271,7 +268,35 @@ public void updated(String factoryPid, Dictionary<String, ?> dictionary)
 	}
 
 	@Activate
-	protected void activate(BundleContext bundleContext) {
+	protected void activate(BundleContext bundleContext)
+		throws InvalidSyntaxException {
+
+		_bundleContext = bundleContext;
+
+		_destinationEventListenerServiceTracker = new ServiceTracker<>(
+			bundleContext,
+			bundleContext.createFilter(
+				"(&(destination.name=*)(objectClass=" +
+					DestinationEventListener.class.getName() + "))"),
```

Comment:
@tinatian Neither `DestinationEventListener` nor `MessageBusEventListener` has any implementation that is registered as service. Since message bus belongs to us, could you decide whether we can get rid of these extension points?